### PR TITLE
Feature/76 locator tool

### LINKED ISSUE #76: Implement Locator Tool
[Locator](https://developers.arcgis.com/javascript/latest/api-reference/esri-rest-locator.html)

1. Implement a C# passthrough `LogicComponent` with all methods.
2. Create a samples page to demonstrate some or all functionality.

### PR DESCRIPTION
**Pull request by AndersenBell**
Closes #76 
Implementing https://developers.arcgis.com/javascript/latest/api-reference/esri-rest-locator.html

Couple things:

1. Overloaded the calls with a default URL: https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer

2. When I was testing, I needed more permissions for the AddressesToLocations web call. I added a note to the calls that require it to make sure they add the permission for their API token.

### CODE REVIEW

File: src/dymaptic.GeoBlazor.Core/Model/Locator.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+﻿using dymaptic.GeoBlazor.Core.Components.Geometries;
+using dymaptic.GeoBlazor.Core.Objects;
+using Microsoft.JSInterop;
+
+namespace dymaptic.GeoBlazor.Core.Model;
+
+/// <summary>
+///    Represents a geocode service resource exposed by the ArcGIS Server REST API. It is used to generate candidates for an address. It is also used to generate batch results for a set of addresses.
+///    Set the URL to the ArcGIS Server REST resource that represents a Locator service, for example: https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer.
```
Comment (seahro) [inline src/dymaptic.GeoBlazor.Core/Model/Locator.cs:9]: Per the discussion.  The doing batch item geocoding will need the user to have an api key setup to do those geocoding calls. reference to the developer.arcgis.com and setting up the api key to have the permissions set to "Geocode (Stored)" vs the default setting... Geocode (Not Stored)

image

Comment (AndersenBell) [reply src/dymaptic.GeoBlazor.Core/Model/Locator.cs:9]: added comments to the locator class

File: samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/LocationMethods.razor [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        {
+            <div>@location?.Address  @location?.Score</div>
+            <br />
+        }
+    </label>
+</div>
+<div class="form-group" style="border: 1px solid black; padding: 0.5rem;">
+
+    <h2>Address To Locations method</h2>
+    <code>var locations = await Location.AddressToLocations(new Address("380 New York Street", "Redlands", "CA", "92373"), outFields:new(){"*"});</code>
```
Comment (seahro) [inline samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/LocationMethods.razor:28]: i think we are missing `_locations1` here.

**Review by seahro — APPROVED** [review]
Looks good, one thing...see note in `LocationMethods.razor`

**Review by TimPurdum — CHANGES_REQUESTED** [review]
Unfortunately, some of this was already implemented in GeoBlazor Pro, and should be combined with that, rather than re-implemented here. Sorry for the confusion on the open gissue. That's my fault for not updating gissues. We can talk about how to proceed and use this implementation.

**Comment by TimPurdum** [comment]
Re-opening. I had implemented this in Pro, but it was correctly listed as a future feature for GeoBlazor Core. @AndersenBell , before we merge this, I'm going to ask you to review the Pro implementation and move anything useful into this implementation, then remove the Pro version.

File: samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/LocationMethods.razor (hunk 2 above)
Comment (AndersenBell) [reply samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/LocationMethods.razor:28]: this is just the example, so it can be simple locations

**Comment by TimPurdum** [comment]
I merged the pro locator functions with this implementation. Also re-targeted to merge to the version/3.0.0 branch.

**Review by TimPurdum — APPROVED** [review]

**Pull request merged**